compliance/index/v4/node.go

```diff
@@ -0,0 +1,182 @@
+package v4
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/quay/claircore"
+	ccindexer "github.com/quay/claircore/indexer"
+	"github.com/quay/claircore/indexer/controller"
+	"github.com/quay/claircore/rhel"
+	rpm2 "github.com/quay/claircore/rpm"
+	"github.com/quay/zlog"
+	"github.com/stackrox/rox/compliance/collection/compliance"
+	"github.com/stackrox/rox/compliance/collection/intervals"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/scannerv4/mappers"
+)
+
+// The layer carries a hardcoded digest, as it is exclusively used for passing
+// ClairCore checks, not for Scanners matching
+var layerDigest = fmt.Sprintf("sha256:%s", strings.Repeat("a", 64))
+
+type localNodeIndexer struct {
+}
+
+// NewNodeIndexer creates a new node indexer
+func NewNodeIndexer() compliance.NodeIndexer {
+	return &localNodeIndexer{}
+}
+
+// GetIntervals
+func (l *localNodeIndexer) GetIntervals() *intervals.NodeScanIntervals {
+	i := intervals.NewNodeScanIntervalFromEnv()
+	return &i
+}
+
+// IndexNode indexes a live fs.FS at the container mountpoint given in the basePath.
+func (l *localNodeIndexer) IndexNode(ctx context.Context) (*v4.IndexReport, error) {
+	report := &claircore.IndexReport{
+		Packages:      map[string]*claircore.Package{},
+		Environments:  map[string][]*claircore.Environment{},
+		Distributions: map[string]*claircore.Distribution{},
+		Repositories:  map[string]*claircore.Repository{},
+		Files:         map[string]claircore.File{},
+	}
+
+	layer, err := constructLayer(ctx, layerDigest, env.NodeScanningV4HostPath.Setting())
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if tmpErr := layer.Close(); tmpErr != nil {
+			err = tmpErr
+		}
+	}()
+
+	reps, err := runRepositoryScanner(ctx, layer)
+	if err != nil {
+		return nil, err
```

Comment:
I would be cool to wrap those errors, as we have many of them one by one and later we would like to know which if-block produced that error

Reply:
I wrap the errors inside the private functions, so we already have a custom error, in this case [this one](https://github.com/stackrox/stackrox/pull/12227/files/5f2f225a2f68051440ce6b755d12c81fe9d5163f#diff-6fc1ea8e64a161479830ac48435b16e14ff2aca9b3e2884b04ce3315b184945dR116). I opted to not add another layer of wrapping to this to keep the messages short.